### PR TITLE
feat: visual update reminder for admins

### PR DESCRIPTION
**Pull request by A-Daneel**

### CODE REVIEW

**Comment by A-Daneel** [comment]
@jasperweyne on the user end a small indicator will appear to indicate an update is available.
When accessing the admin panel, you'll be greeted with the following screen instead of the "activity gray cards".
afbeelding

I feel this is a nice middle ground, not too intrusive yet it seeing the days count up will motivate an user to update.

**Comment by jasperweyne** [comment]
I like the concept, although I think that it can be more refined visually. My suggestion would be to have the notice be in some sort of box or divided in another way, and leave the activity blocks visible. Maybe reword the notice to "a new version has been available since" and turn the link to the updater into a button. Furthermore, I see that some modifications were made to the main layout as well, I'm curious to what this would look like. If you're able, please share a screenshot of that as well!

File: src/Template/UpdateChecker.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            if (null === $httpClient) {
+                throw new \InvalidArgumentException('An explicit HttpClient must be provided during testing.');
+            }
+            $this->client = $httpClient;
+        } else {
+            $this->client = new CachingHttpClient(
+                HttpClient::create(),
+                new Store("{$kernel->getCacheDir()}/releases")
+            );
+        }
```
Comment (jasperweyne) [inline src/Template/UpdateChecker.php:29]: This code now behaves differently in a test environment. To a certain extent, that's desirable: we want to get an exception in a testing environment if no explicit httpClient is provided. However, I'd argue that this different behaviour should be limited to notices. Therefore my suggestion would be to restructure this to
```php
        if (null === $httpClient) {
            if ('test' === $kernel->getEnvironment()) {
                throw new \InvalidArgumentException('An explicit HttpClient must be provided during testing.');
            }
            $this->client = $httpClient;
        } else {
            $this->client = new CachingHttpClient(
                HttpClient::create(),
                new Store("{$kernel->getCacheDir()}/releases")
            );
        }
        ```

**Review by jasperweyne — CHANGES_REQUESTED** [review]

File: src/Template/UpdateChecker.php (hunk 1 above)
Comment (jasperweyne) [reply src/Template/UpdateChecker.php:29]: You're right, I see now how my suggestion is flawed. However, I feel the logic to change things still stands. Therefore, I propose the following:
```php
        if (null === $httpClient) {
            if ('test' === $kernel->getEnvironment()) {
                throw new \InvalidArgumentException('An explicit HttpClient must be provided during testing.');
            }
            $this->client = new CachingHttpClient(
                HttpClient::create(),
                new Store("{$kernel->getCacheDir()}/releases")
            );
        } else {
            $this->client = $httpClient;
        }
```
With this structure, it should behave similar to your code, with the notable exception of calling the constructor with a HttpClient in any other environment than test; with the current code it would be ignored, while this structure always allows for a prepopulated value.

**Review by jasperweyne — APPROVED** [review]
